src/Features/Blockcore.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -1271,19 +1271,20 @@ public IActionResult RemoveTransactions([FromQuery]RemoveTransactionsModel reque
                 }
 
                 // If the user chose to resync the wallet after removing transactions.
-                if (result.Any() && request.ReSync)
+                if (request.ReSync)
                 {
-                    // From the list of removed transactions, check which one is the oldest and retrieve the block right before that time.
-                    DateTimeOffset earliestDate = result.Min(r => r.creationTime);
-                    ChainedHeader chainedHeader = this.chainIndexer.GetHeader(this.chainIndexer.GetHeightAtTime(earliestDate.DateTime));
+                    Wallet wallet = this.walletManager.GetWallet(request.WalletName);
+
+                    // Initiate the scan one day ahead of wallet creation.
+                    // If the creation time is DateTime.MinValue, don't remove one day as that throws exception.
+                    ChainedHeader chainedHeader = this.chainIndexer.GetHeader(this.chainIndexer.GetHeightAtTime(wallet.CreationTime.DateTime != DateTime.MinValue ? wallet.CreationTime.DateTime.AddDays(-1) : wallet.CreationTime.DateTime));
 
                     // Update the wallet and save it to the file system.
-                    Wallet wallet = this.walletManager.GetWallet(request.WalletName);
                     wallet.SetLastBlockDetails(chainedHeader);
                     this.walletManager.SaveWallet(wallet);
 
-                    // Start the syncing process from the block before the earliest transaction was seen.
-                    this.walletSyncManager.SyncFromHeight(chainedHeader.Height - 1);
+                    // Start the sync from the day before it was created.
+                    this.walletSyncManager.SyncFromHeight(chainedHeader.Height);
```

Comment:
why remove the -1?

Reply:
It previously synced just minus 1 block, now it syncs minus 1 day, see 3rd line from above this line.